### PR TITLE
fix: prime task create last-used cache before dialog effects

### PR DESCRIPTION
**Pull request by carlosflorencio**
The create task dialog could read an empty last-used cache before backend settings were mirrored locally, so selectors fell back to defaults. The cache now primes before child dialog effects run, keeping the last repo, branch, executor, and agent choices intact.

## Important Changes

- Use a layout effect to mirror backend task-create defaults before dialog selection effects run.
- Harden license generation so go-licenses failures reuse valid committed Go entries with a stale marker instead of dropping the Go ecosystem from the manifest.

## Validation

- `pnpm --filter @kandev/web test -- --run components/state-provider.test.tsx`
- `pnpm --filter @kandev/web test -- --run components/task-create-dialog-effects.test.ts components/task-create-dialog-effects-executor.test.ts components/task-create-dialog-handlers.test.ts`
- `pnpm --filter @kandev/web test -- --run scripts/generate-licenses-fallback.test.ts scripts/generate-licenses.test.ts components/settings/system/licenses-list.test.tsx components/state-provider.test.tsx`
- `pnpm --filter @kandev/web licenses:gen` verified the fallback writes 84 stale Go entries; generated JSON restored afterward.
- `make fmt`
- `make typecheck test lint`
- GitHub CI: Frontend, CodeQL, Preview Environment, E2E shards, and E2E container shards passed.

## Possible Improvements

Low risk: Playwright coverage was not expanded because the dialog behavior change is effect-ordering state sync covered by component tests.

## Checklist

- [ ] I have performed a self-review of my code.
- [ ] I have manually tested my changes and they work as expected.
- [ ] My changes have tests that cover the new functionality and edge cases.
- [ ] If my change touches UI files (`apps/web/`), I have added or updated Playwright e2e tests in `apps/web/e2e/` and verified them with `make test-e2e`.

## Preview Environment

| | |
|---|---|
| **URL** | https://kandev-pr-1509-bwo7.sprites.app |
| **Commit** | `c7a972e` |
| **Agent** | Mock agent |

### CODE REVIEW

File: apps/web/scripts/generate-licenses.ts [hunk 1]
```diff
@@ -216,13 +216,68 @@ function collectGoEntries(): LicenseEntry[] {
   } catch (err) {
```
[the remark this answers is not in the document]
Comment (carlosflorencio) [reply apps/web/scripts/generate-licenses.ts:217]: Addressed in c7a972e. Reused Go entries are now marked with stale: true, the licenses page shows a stale-data warning when those markers are present, and the system-page spec documents the fallback behavior.

[the remark this answers is not in the document]
Comment (carlosflorencio) [reply apps/web/scripts/generate-licenses.ts:217]: Addressed in c7a972e. The fallback cascade is now a pure recoverGoEntriesAfterReportFailure helper with tests for stdout recovery winning, existing-manifest fallback, and no-data returning null.

File: apps/web/scripts/generate-licenses-fallback.test.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import { describe, expect, it } from "vitest";
+
+import { readGoEntriesFromManifest } from "./generate-licenses";
+
+describe("readGoEntriesFromManifest", () => {
+  it("returns valid Go entries from an existing manifest", () => {
```
[the remark this answers is not in the document]
Comment (carlosflorencio) [reply apps/web/scripts/generate-licenses-fallback.test.ts:11]: Addressed in c7a972e. Added explicit tests for invalid and non-array JSON, npm-only manifests, and malformed Go entries being filtered out.

File: apps/web/scripts/generate-licenses.ts [hunk 3]
```diff
@@ -216,13 +216,68 @@ function collectGoEntries(): LicenseEntry[] {
...
+      return existingEntries;
+    }
+    throw new Error("go-licenses failed and no existing Go entries are available");
   } finally {
     rmSync(tmplPath, { force: true });
   }
 
-  const rows = parseGoRows(raw);
-  return rows.filter(isThirdPartyGoRow).map(toGoEntry);
+  return goEntriesFromReport(raw);
```
[the remark this answers is not in the document]
Comment (carlosflorencio) [reply apps/web/scripts/generate-licenses.ts:238]: Addressed in c7a972e. I extracted the cascade into recoverGoEntriesAfterReportFailure and added direct tests for the branch order and empty-result behavior.

File: apps/web/scripts/generate-licenses.ts [hunk 4]
```diff
@@ -216,13 +216,68 @@ function collectGoEntries(): LicenseEntry[] {
...
+        `[licenses] warning: recovered ${recoveredEntries.length} Go entries from go-licenses stdout.\n`,
+      );
+      return recoveredEntries;
+    }
+    const existingEntries = readExistingGoEntries();
+    if (existingEntries.length > 0) {
+      process.stderr.write(
+        `[licenses] warning: reusing ${existingEntries.length} Go entries from the existing manifest.\n`,
+      );
+      return existingEntries;
```
[the remark this answers is not in the document]
Comment (carlosflorencio) [reply apps/web/scripts/generate-licenses.ts]: Addressed in c7a972e. The fallback no longer treats reused Go entries as fresh: it marks them stale, the licenses UI warns when stale Go entries are present, and the spec now documents the behavior. The generator also still recovers fresh stdout rows first and only falls back when no report rows are available.
